src/schema/v2/discoverArtworks.ts

```diff
@@ -0,0 +1,59 @@
+import { GraphQLString, GraphQLFieldConfig, GraphQLInt } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { artworkConnection } from "./artwork"
+import { connectionFromArray } from "graphql-relay"
+import { pageable } from "relay-cursor-paging"
+import gql from "lib/gql"
+import uuid from "uuid/v5"
+
+export const generateUuid = (userId: string) => {
+  if (!userId) return ""
+  return uuid(userId, uuid.DNS).toString()
+}
+
+export const DiscoverArtworks: GraphQLFieldConfig<void, ResolverContext> = {
+  type: artworkConnection.connectionType,
+  args: pageable({
+    userId: { type: GraphQLString },
+    limit: { type: GraphQLInt },
+  }),
+  resolve: async (_root, args, { weaviateGraphqlLoader, artworksLoader }) => {
+    if (!weaviateGraphqlLoader) return
+
+    const { userId, limit = 5 } = args
+
+    const userUUID = generateUuid(userId)
+
+    // TODO: Understand why localhost works here and weaviate://weaviate.stg.artsy.net doesn't
+    const beacon = `weaviate://localhost/InfiniteDiscoveryUsers/${userUUID}`
```

Comment:
Need to look into this more...

Reply:
Curious about this. I've seen internal hostname references (like `weaviate-internal`) but not routing _by protocol_. How does this work?

Reply:
I'm not sure I have much more to add, aside from this resource: [Weaviate GraphQL Search Operators](https://weaviate.io/developers/weaviate/api/graphql/search-operators#variables-1).

<img width="1024" alt="image" src="https://github.com/user-attachments/assets/a9ca5002-b2e6-45bb-b2b6-6e9519e59546">
From my understanding, if we have a multi-tenant strategy, it could be useful as an identifier. But I could be completely wrong about that.